src/components/overview/overview.jsx

```diff
@@ -1,31 +1,56 @@
-import React from 'react';
+import React, {useState} from 'react';
 import styles from './overview.module.css'
 import ImageGallery from './ImageGallery.jsx';
 import ProductInfo from './ProductInfo.jsx'
 import Description from './Description.jsx';
 import Features from './Features.jsx'
+import json from './styleSample.json'
+import {useProductContext} from '../../context/ProductContext.jsx'
+//Will pass in productcontext.style to to style context as default value
+
+
+export const StyleIndexContext = React.createContext({
+  styleIndex:0,
+  changeStyleIndex: () => {},
+});
+
+//Refactoring to pass Style info as props to components
+// Then the StyleContext will change the Index for the ImageGallery and AddToCart to determine
+  //The selected style
+
+//If new productInfo is provide,
+//the style index will be reset to first index.
+//Maybe create state for overview to retain the state of style index, and cureent product info
+  //Could compare to previous ID to see if there needs to be an index reset
 
 const Overview = (props) => {
 
+    const [styleIndex, setStyleIndex] = useState(0);
+    var product = useProductContext();
+
+    console.log(styleIndex)
     return(
-      <div className={styles.widget}>
+      <StyleIndexContext.Provider value ={{changeStyleIndex : setStyleIndex}}>
```

Comment:
I just want to make sure I'm understanding this...

So what you are passing down to all the consumers is a function for changing the styleIndex state of Overview?

I'm confused on how StyleIndexContext ties into it because because you are passing to React.createContext() ties into this... I had to go look up the [docs](https://reactjs.org/docs/context.html#:~:text=The%20defaultValue%20argument,to%20use%20defaultValue.) for React.createContext() and from what I understand you pass in a default value that is used if a consumer of this context doesn't have a matching provider above it in the tree. (So most of the time the default object you gave this context doesn't matter)

I guess it's getting clearer now. You only need the consumers to be able to change the styleIndex state and when it gets changed then the overview will re-render because it's state is being changed... Looks good!